portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -11,22 +11,43 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
 package com.google.sps.servlets;
-
+ 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import com.google.gson.Gson;
+import java.util.ArrayList;
+ 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
+  private final ArrayList<String> comments = new ArrayList<String>();
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jenny Sheng!</h1>");
+    comments.clear();
```

Comment:
Since those are hardcoded, there is no need to remove and recreate them on every request. On the other hand, you will likely replace this with datastore-sourced data soon.

I would either initialize `comments` with hardcoded values when initilaizing it (`Arrays.asList()` is the easiest way to initialize a list inline) or turn `comments` into a local variable. I'd favor the latter because HTTP handlers should be _stateless_ -- they should behave the same way for each identical request. As it happens, neither `ArrayList` nor `WebServlet` are thread-safe, so if you spawn multiple simultaneous requests to `/data`, you might see some very interesting behavior!